File: first_recursion/firstRecursion.js
const recursion = (arr) => {
    const hashMap = {};

    for (let i = 0; i < arr.length; i++) {
        if (!(hashMap[arr[i]])) {
            hashMap[arr[i]] = 1
        } else {
            return arr[i]
        }
    }
    return undefined
}

const array = [1, 2, 4, 5, 3, 7, 8];

recursion(array)

File: first_recursion/first_recursion.rb
def recursion(arr) 
  mapping = {}

  arr.each do |ele|
    if !(mapping[ele])
      mapping[ele] = true
    else 
      return ele
    end
  end
  return nil
end


array = [1,2,4,5,3,7,8,12,4,12];

recursion(array)